feature/account/setup/src/main/kotlin/app/k9mail/feature/account/setup/ui/validation/AccountValidationScreen.kt

```diff
@@ -0,0 +1,95 @@
+package app.k9mail.feature.account.setup.ui.validation
+
+import androidx.activity.compose.BackHandler
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.res.stringResource
+import app.k9mail.core.ui.compose.common.DevicePreviews
+import app.k9mail.core.ui.compose.common.mvi.observe
+import app.k9mail.core.ui.compose.designsystem.template.Scaffold
+import app.k9mail.core.ui.compose.theme.K9Theme
+import app.k9mail.core.ui.compose.theme.ThunderbirdTheme
+import app.k9mail.feature.account.common.ui.AppTitleTopHeader
+import app.k9mail.feature.account.common.ui.WizardNavigationBar
+import app.k9mail.feature.account.setup.R
+import app.k9mail.feature.account.setup.ui.validation.AccountValidationContract.Effect
+import app.k9mail.feature.account.setup.ui.validation.AccountValidationContract.Event
+import app.k9mail.feature.account.setup.ui.validation.AccountValidationContract.ViewModel
+import com.fsck.k9.mail.server.ServerSettingsValidationResult
+
+@Composable
+internal fun AccountValidationScreen(
+    onNext: () -> Unit,
+    onBack: () -> Unit,
+    viewModel: ViewModel,
+    modifier: Modifier = Modifier,
+) {
+    val (state, dispatch) = viewModel.observe { effect ->
+        when (effect) {
+            is Effect.NavigateNext -> onNext()
+            is Effect.NavigateBack -> onBack()
+        }
+    }
+
+    LaunchedEffect(key1 = Unit) {
+        dispatch(Event.ValidateServerSettings)
```

Comment:
Will this kick off an additional server settings validation when the activity is restarted e.g. due to a configuration change?

Reply:
Yes, it is triggered once there was a configuration change and there is a check for isSuccess to avoid runing the check again.